### ISSUE TITLE
Error when updating on PHP 8

### ISSUE THREAD

**Issue by SharkyKZ** [joomla/joomla-cms#31308]
### Steps to reproduce the issue

Try to update on PHP 8.

### Expected result

Works.

### Actual result

```
ERROR:
Invalid AJAX data:
<br />
<b>Fatal error</b>:  Uncaught ValueError: Path cannot be empty in C:\wamp\www\joomla_39\administrator\components\com_joomlaupdate\restore.php:879
Stack trace:
#0 C:\wamp\www\joomla_39\administrator\components\com_joomlaupdate\restore.php(879): fopen('', 'rb')
#1 [internal function]: AKAbstractUnarchiver-&gt;__wakeup()
#2 C:\wamp\www\joomla_39\administrator\components\com_joomlaupdate\restore.php(6678): unserialize('O:9:&quot;AKFactory&quot;...')
#3 C:\wamp\www\joomla_39\administrator\components\com_joomlaupdate\restore.php(6724): AKFactory::getInstance('O:9:&quot;AKFactory&quot;...')
#4 C:\wamp\www\joomla_39\administrator\components\com_joomlaupdate\restore.php(8120): AKFactory::unserialize('O:9:&quot;AKFactory&quot;...')
#5 C:\wamp\www\joomla_39\administrator\components\com_joomlaupdate\restore.php(8207): masterSetup()
#6 {main}
  thrown in <b>C:\wamp\www\joomla_39\administrator\components\com_joomlaupdate\restore.php</b> on line <b>879</b><br />
```

### System information (as much as possible)

PHP 8.0.0 RC3.

**Comment by stAn47** [comment]
a few more errors collected on php8.0 vs 3.9.22 

```

core readmore plugin gives fatal error due to incorrect parameters compared to passed named parameters 
Error message data: 
1 Unknown named parameter $name in file: /libraries/joomla/event/event.php line: 70 

Backtrace: 
/libraries/joomla/event/event.php:70
#0 /libraries/src/Editor/Editor.php(396): JEvent->update()
#1 /plugins/editors-xtd/readmore/readmore.php(44): Joomla\CMS\Editor\Editor->getContent()

non core JCE - strict parameters checks should be handled by plugin: 
Error message data: 
1 Unknown named parameter $params in file: /libraries/joomla/event/event.php line: 70 


Backtrace: 
/libraries/joomla/event/event.php:70
#0 /libraries/src/Editor/Editor.php(326): JEvent->update()
#1 /libraries/src/Form/Field/EditorField.php(256): Joomla\CMS\Editor\Editor->display()


JText::sprintf:

Error message data: 
1 Unknown format specifier "S" in file: /libraries/src/Language/Text.php line: 321 

//for strings of %2$ immidiatelly followed by character - %2$String


php.ini from older PHP version: 
//if php.ini uses 60s this will give fatal error:
@set_time_limit(ini_get('max_execution_time'));

Joomla update 3.9.22 to 3.9.22 on php8 (on php7.4 this is ok)

ValueError: Path cannot be empty /administrator/copmonetns/com_joomlaupdate/restore.php line  879



```<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31308">issues.joomla.org/tracker/joomla-cms/31308</a>.</sub>

**Comment by SharkyKZ** [comment]
@stAn47 See PR https://github.com/joomla/joomla-cms/pull/30608 for first issue.

>//for strings of %2$ immidiatelly followed by character - %2$String

This was not valid format in the first place and I can't find any instances of this in core language files. But this could be present in language packs and 3rd party extension language files.

**Comment by richard67** [comment]
Please test pull request #31442 for the `Uncaught ValueError: Path cannot be empty in /administrator/components/com_joomlaupdate/restore.php:879` issue.

@SharkyKZ Could you have a brief look on y my PR? If it is ok, shall we close this issue here, or shall we leave it open due to other findings reported by @stAn47 above?

**Comment by HLeithner** [comment]
Fixed in #31442
